lib/Sema/CSSimplify.cpp

```diff
@@ -2972,6 +2972,11 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       
     } else {
       if (!hasStaticMembers) {
+        if (cand->isImplicit() && isa<ConstructorDecl>(cand)) {
```

Comment:
I think a better fix is to not synthesize a constructor at all in this case.

In other words, the error comes up because we're type checking the body of a synthesized method, and this type checking fails -- so let's just not synthesize invalid code at all.